funding/commitment_type_negotiation.go

```diff
@@ -353,15 +379,39 @@ func implicitNegotiateCommitmentType(local,
 	return &chanType, lnwallet.CommitmentTypeLegacy
 }
 
-// hasFeatures determines whether a set of features is supported by both the set
-// of local and remote features.
+// keeps track of the feature bit mismatches between the local and remote nodes.
+type FeaturesBitDiff struct {
+	local  []lnwire.FeatureBit
+	remote []lnwire.FeatureBit
+}
+
+// isEmpty returns true if the local and remote features are empty arrays.
+func (f *FeaturesBitDiff) isEmpty() bool {
+	return len(f.local) == 0 && len(f.remote) == 0
+}
+
+// String returns the string version of the struct FeaturesBitDiff.
+func (f FeaturesBitDiff) String() string {
+	return fmt.Sprintf("local %v - remote %v", f.local, f.remote)
+}
+
+// hasFeatures determines whether a set of features is supported by both the
+// local and remote features. A FeaturesBitDiff struct is returned with
+// the missing feature bits for each side.
 func hasFeatures(local, remote *lnwire.FeatureVector,
-	features ...lnwire.FeatureBit) bool {
+	features ...lnwire.FeatureBit) (bool, FeaturesBitDiff) {
```

Comment:
I probably would just return `FeaturesBitDiff` instead of `(bool, FeaturesBitDiff)` since we can call `missing.isEmpty()` at the other places to replace the bool `ok`, which seems cleaner.

Reply:
make sense

Reply:
@yyforyongyu do you want to rename also the function?

Reply:
gently pinging again @yyforyongyu to avoid another iteration about the name inside the review

Reply:
yeah I probably would name it `checkFeaturesDiff` and return the diff.

Reply:
and add a unit test for this method.